source/common/event/dispatcher_impl.cc

```diff
@@ -153,8 +155,28 @@ DispatcherImpl::createClientConnection(Network::Address::InstanceConstSharedPtr
                                        Network::TransportSocketPtr&& transport_socket,
                                        const Network::ConnectionSocket::OptionsSharedPtr& options) {
   ASSERT(isThreadSafe());
-  return std::make_unique<Network::ClientConnectionImpl>(*this, address, source_address,
-                                                         std::move(transport_socket), options);
+
+  auto address_type_name = [](const Network::Address::InstanceConstSharedPtr& addr) {
+    ASSERT(addr != nullptr);
+    switch (addr->type()) {
```

Comment:
I wonder if we should add getter for the client connection factory into the `Network::Address::Instance`

Reply:
done